app/controllers/guide_progress_controller.rb

```diff
@@ -1,7 +1,18 @@
 class GuideProgressController < ApplicationController
+  before_action :check_not_in_exam!, only: :destroy
+
   def destroy
-    guide = Guide.find(params[:guide_id])
     guide.clear_progress! current_user
```

Comment:
Shouldn't we change this to @guide?

Reply:
no?

Reply:
Please ignore this comment. I did not see the corresponding method